File: src/com/Torvald/JsonFetcher.java
package com.Torvald;

import com.google.gson.JsonObject;
import com.google.gson.JsonParser;

import java.io.IOException;
import java.nio.file.FileSystems;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.Iterator;

/**
 * Created by minjaesong on 16-02-15.
 */
public class JsonFetcher {

    private static StringBuffer jsonString;

    public static JsonObject readJson(String jsonFilePath) throws IOException {
        jsonString = new StringBuffer(); // reset buffer every time it called
        readJsonFileAsString(jsonFilePath);

        JsonParser jsonParser = new JsonParser();
        JsonObject jsonObj = jsonParser.parse(jsonString.toString()).getAsJsonObject();

        return jsonObj;
    }

    public static ArrayList<String> readJsonAsString(String jsonFilePath) throws IOException {
        ArrayList<String> jsonFileLines = new ArrayList<>();
        Files.lines(
                FileSystems.getDefault().getPath(jsonFilePath)
        ).forEach(jsonFileLines::add);
        return jsonFileLines;
    }

    private static void readJsonFileAsString(String path) throws IOException {
        Files.lines(
                FileSystems.getDefault().getPath(path)
        ).forEach(jsonString::append); // JSON does not require line break
    }
}

File: src/net/torvald/JsonFetcher.kt
package net.torvald

import com.google.gson.JsonObject
import com.google.gson.JsonParser
import java.io.File

import java.io.IOException
import java.nio.file.FileSystems
import java.nio.file.Files
import java.util.ArrayList
import java.util.function.Consumer

/**
 * Created by minjaesong on 16-02-15.
 */
object JsonFetcher {

    private var jsonString: StringBuffer? = null

    @Throws(IOException::class)
    fun readJson(jsonFilePath: String): JsonObject {
        jsonString = StringBuffer() // reset buffer every time it called
        readJsonFileAsString(jsonFilePath)

        println("Reading JSON $jsonFilePath")

        val jsonParser = JsonParser()
        val jsonObj = jsonParser.parse(jsonString!!.toString()).asJsonObject

        return jsonObj
    }

    @Throws(IOException::class)
    fun readJson(jsonFile: File): JsonObject {
        jsonString = StringBuffer() // reset buffer every time it called
        readJsonFileAsString(jsonFile.canonicalPath)

        println("Reading JSON ${jsonFile.path}")

        val jsonParser = JsonParser()
        val jsonObj = jsonParser.parse(jsonString!!.toString()).asJsonObject

        return jsonObj
    }

    @Throws(IOException::class)
    private fun readJsonFileAsString(path: String) {
        Files.lines(FileSystems.getDefault().getPath(path)).forEach(
                { jsonString!!.append(it) }
        ) // JSON does not require line break
    }
}
